packages/legend-graph/src/helpers/DomainHelper.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Class } from '../models/metamodels/pure/packageableElements/domain/Class';
+
+export const isVersionedClass = (val: Class): boolean => {
+  for (let i = 0; i < val.stereotypes.length; i++) {
+    const stereotype = val.stereotypes[i]?.value.value;
+    if (
+      stereotype === 'bitemporal' ||
+      stereotype === 'businesstemporal' ||
+      stereotype === 'processingtemporal'
+    ) {
+      return true;
+    }
+  }
+  for (let i = 0; i < val.generalizations.length; i++) {
+    const superType = val.generalizations[i]?.value.rawType;
```

Comment:
Instead of this will using the `.some` method on the supertype array be better instead of the for loop. 
Same goes for the logic above.